src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -460,7 +460,9 @@ private void clearSuggestions() {
 		setSuggestions(null);
 
 		if (currentInputConnection != null) {
-			textField.setComposingText("");
+			if (textField != null) {
```

Comment:
`textField` and `inputType` are assumed to be defined everywhere. Instead of this `if`, let's initialize them at the beginning around line 30:
```java
	private TextField textField = new TextField(null, null);
	private InputType inputType = new InputType(null, null);
```

All their methods just do nothing when `InputConnection` and `EditorInfo` are null, which means, this operation will be safe.